src/utils/span-pointers.spec.ts

```diff
@@ -0,0 +1,159 @@
+import { getSpanPointerAttributes } from "./span-pointers";
+import { eventTypes } from "../trace/trigger";
+
+// tslint:disable-next-line:no-var-requires
+const { S3_PTR_KIND, SPAN_POINTER_DIRECTION } = require("dd-trace/packages/dd-trace/src/constants");
+// tslint:disable-next-line:no-var-requires
+const util = require("dd-trace/packages/dd-trace/src/util");
+
+// Mock the external dependencies
+jest.mock("./log", () => ({
+  logDebug: jest.fn(),
+}));
+
+interface SpanPointerAttributes {
+  pointerKind: string;
+  pointerDirection: string;
+  pointerHash: string;
+}
+
+describe("span-pointers utils", () => {
+  const mockPointerHash = "mock-hash-123";
+
+  beforeEach(() => {
+    jest.spyOn(util, "generatePointerHash").mockReturnValue(mockPointerHash);
```

Comment:
why not let the generate pointer hash function actually run?

Reply:
I already have tests for `generatePointerHash` in the tracer; i just wanted to test only the logic behind the functions created in this PR. That way, if we ever make changes to how the `generatePointerHash` function works in the tracer, it won't break the tests here.